mod/youtubevideo/renderer.php

```diff
@@ -80,4 +99,61 @@ public function footer()
     {
         return $this->output->footer();
     }
+
+    /**
+     * Render The Manage Link
+     * 
+     * @param int $courseid
+     * @param context $context
+     * @return string HTML
+     */
+    public function manage_link($courseid, $context)
+    {
+        if (!has_capability('mod/youtubevideo:manage', $context)) {
+            return '';
+        }
+
+        $buttonAttributes = [
+            'class' => 'btn btn-primary youtubevideo-manage-btn',
+            'role' => 'button'
+        ];
+
+        $buttonContent = $this->get_icon('fa-cog') . ' ' . $this->get_string('manageyoutubevideos');
+        $manageUrl = $this->get_manage_url($courseid);
+
+        $button = $this->create_button($manageUrl, $buttonContent, $buttonAttributes);
+
+        return $this->wrap_in_div($button, 'youtubevideo-manage-button');
+    }
+
+    private function get_icon($iconClass)
+    {
+        $icon = html_writer::tag('i', '', ['class' => "fa $iconClass", 'aria-hidden' => 'true']);
+
+        return $icon;
+    }
+
+    private function get_string($identified)
+    {
+        return get_string($identified, 'youtubevideo');
+    }
+
+    private function get_manage_url($courseid)
+    {
+        $manageUrl = new moodle_url('/mod/youtubevideo/manage.php', ['id' => $courseid]);
+
+        return $manageUrl;
+    }
+
+    private function create_button($url, $content, $attributes)
+    {
+        $button = html_writer::link($url, $content, $attributes);
+
+        return $button;
+    }
+
+    private function wrap_in_div($content, $class)
+    {
+        return html_writer::div($content, $class);
+    }
```

Comment:
Est-ce que tu as vraiment besoin de créer ces fonctions ?
Un ⤵️
```suggestion
$button = $this->output->action_link($manageUrl, $buttonContent, null, ['class' => 'btn btn-primary youtubevideo-manage-btn']); 
```
aurait pu suffire.

Reply:
J'ai créé ces functions pour rendre la function principale plus légère et facilité la maintenance